clip_benchmark/models/transformers_clip.py

```diff
@@ -0,0 +1,24 @@
+from torch import nn
+from transformers import AutoModel, AutoProcessor
+from functools import partial
+
+class TransformerWrapper(nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.model = model
+    
+    def encode_text(self, text):
+        return self.model.get_text_features(**text)
+
+    def encode_image(self, image):
+        return self.model.get_image_features(image["pixel_values"].squeeze(1))
```

Comment:
Interestingly, why should we remove 1-st dim?

Reply:
I wanted to brain storm on this bit, thanks for asking this question.
The load function should return a model, transforms, and a tokenizer. In the load function I have written it returns the model, the image processor, and the tokenizer.

The transform is used in the collation function while building the dataloader, which is where it adds another dimension to the tensors. So I noticed that the `images["pixel_values].shape=(b, 1, c, h, w)`.

Is there a way I could extract the `transform` function from an `image_processor`?